core/templates/dev/head/pages/exploration-player-page/services/exploration-engine.service.ts

```diff
@@ -444,6 +444,13 @@ oppia.factory('ExplorationEngineService', [
       },
       isAnswerBeingProcessed: function() {
         return answerIsBeingProcessed;
+      },
+      currentStateName: function() {
+        return visitedStateNames[visitedStateNames.length - 1];
+      },
+      getImageFilePath: function(stateName, imageId) {
```

Comment:
I'm not sure how this method is integrated, can you give a brief info on when and how it's get called?

Reply:
Whenever a user comes on learner view, I extract ```id-with-value``` from image tag,  and then extract file path of an image with the help of this function.
This function is called in learner view, not in editor view since ```ExplorationEngineService``` is a service related to learner view.